### ISSUE TITLE
Polling interval needs variance to avoid being detected as an "attack" by XFI Gateway/"AI"

### ISSUE THREAD

**Issue by kaicherry** [nzapponi/homebridge-simplisafe3#45]
Is this better? :)

After some (more) investigation, actual calls to SimpliSafe, and a look at the routing, I *believe* the issue reported may be related to Comcast customers with xFI hardware, and could happen with other network monitoring schemes.

It appears that due to the regularity + frequency of polling calls (which could effect other software, not just this one), the combination is being detected as a "network attack" against SimpliSafe's IP address. I believe having deviance around the polling interval to SimpliSafe could mitigate this.

**Comment by kaicherry** [comment]
To add to this, I have made a couple of changes to SimpliSafe.js ( a const that adds a 1-3 second random delay) for some of the GET calls. If I find this in fact stops Xci from detecting calls to SimpliSafe as a "threat", I'll submit a PR tomorrow.

-K

**Comment by nikonratm** [comment]
Interesting, a bit surprising because I can only imagine there are other devices doing similar regular polling. Wonder if it’s really just limited to Comcast + xFi or if it’s other networks or even on SS side. Anyway, sounds like a reasonable fix to try. Nice sleuthing!

**Comment by kaicherry** [comment]
Yeah...it *seemed* at first like it was SS, but I called them, we had a chat, they made it clear what they are doing, and we discovered the requests were never getting to them. At our new house, XFi is "new" and the 'Advanced Network Protection' feature appears to be actively monitoring, and blocking traffic. When I noticed the SS IP matched one of the blocked ones, I kind of worked back from there. SS is not blocking, at least not in this instance. By introducing some randomness, I am hopping this...'monitor' leaves it alone. I'll have a better idea later today, as  upped the refresh frequency in addition to adding the randomness to try to "force" a cutoff.

**Comment by nikonratm** [comment]
Did SS tell you any detail about their rate-limiting efforts? There have been lots of people here (and on another Homebridge SS plugin I used to use) who experience some form of rate-limiting, myself included and I imagine most of us are not on Comcast XFi...

**Comment by virtualjad** [comment]
My IP was just blocked again, resulting in `403 Forbidden`. This is happening about once/week now. Another issue is my HB service stops when the block occurs. It comes up fine on reload (even while still blocked). I'm usually able to swap WAN interfaces and get a new public IP as a work-around, but that has its own headaches due to upstream dependancies. 

Initial assumption was SS is blocking rapid queries. I increased the query interval time from 10sec to 15sec and now 30sec. OP mentions xFi as the potential cause but I'm on Verizon FiOS.

**Comment by murphmr** [comment]
I also was having this issue yesterday afternoon. I did call SS and they suggested I wait 30 minutes and try again. I talked to AT&T support but they said that the request was getting to AWS according to the trace route from my home gateway. It seems to me that Amazon AWS ELB could be doing some blocking based on activity or something like that. I think that this might be something SS knows about but doesn't have a fix for yet. 

I stopped HB and the 403 did go away in a few hours.

**Comment by virtualjad** [comment]
Hit me again yesterday. Since i have many other plugins running my only path is to pull this plugin and config out and wait a while before reinstating. It's become a frustrating loop.

**Comment by nzapponi** [comment]
Funny it hits people more or less at the same time - I was blocked last night as well. It must be something to do with their AWS load balancing config. I don't believe there's much we can do though... Anyone else have ideas?

**Comment by murphmr** [comment]
Has anyone else tried to call SS support on this issue? I did once but maybe if we get a few more people to call when it is happening that will convince them it is not an isolated issue. But my concern is would it negatively impact this effort if they found that this project is somehow tied to the issue.

**Comment by graanco** [comment]
> I also was having this issue yesterday afternoon. I did call SS and they suggested I wait 30 minutes and try again. I talked to AT&T support but they said that the request was getting to AWS according to the trace route from my home gateway. It seems to me that Amazon AWS ELB could be doing some blocking based on activity or something like that. I think that this might be something SS knows about but doesn't have a fix for yet.
> 
> I stopped HB and the 403 did go away in a few hours.

So... When I've been testing I did have the 403 block issue happen to me many of times. When you try to log in on the webpage it seems like you are logged in but it challenges you again with an internal user & password. It takes 1 hour to clear it out. It is a pain when developing the port to homebridge. 

Also the UUID for logging in to the web app on this project needs to be changed with the way it works. I have been thinking of getting it to be a little more personal to the end user. This might be another reason there is an issue with 403s. Too many people no the same UUID.

Secondly, I suggest that the polling works differently. There is other ways to do the polling as I did with my code. I am hoping when I get time I am going to look into if there are any more Socket.io response that simple safe has hidden.
